lib/offset_gen.rb

```diff
@@ -0,0 +1,25 @@
+require 'date'
+
+class OffsetGen
+  attr_reader :date
+
+  def initialize(date = Date.today.strftime('%d%m%y'))
+    @date = date
+  end
+
```

Comment:
Here offset gen is also initialized with a default value that can be changed with input